GlobalScaling/GlobalScalingTests.c

```diff
@@ -0,0 +1,175 @@
+#include "GlobalScalingTests.h"
+#include "GlobalScaling.h"
+
+void runGlobalScalingTests(){
+    globalScalingCleanTest();
+
+    globalScalingDirtyTest1();
+
+    globalScalingDirtyTest2();
+
+    globalScalingDirtyTest3();
+
+    globalScalingDirtyTest4();
+
+    globalScalingDirtyTest5(); 
+}
+
+/*
+    -First clean test
+    -intializes each index of the matrix to the value of 1
+    -calls globalscaling
+    -checks each index if the correct value was entered at that index
+    -if unsuccessful prints test failed
+*/
+void globalScalingCleanTest(){
+    for(int i = 0; i < 4; i++){
+        for(int j = 0; j < 4; j++){
+            transformationMatrix[i][j] = 1.0;
+        }
+    }
+    printf("*************************Running clean test****************************");
+    float check = 1.0;
+    globalScaling();
+    for(int i = 0; i < 4; i++){
+        for(int j = 0; j < 4; j++){
+            if(transformationMatrix[i][j] != check){
+                printf("\n");
+                printf("Global scaling: globalScalingCleanTest failed\n");
```

Comment:
this gets printed even though the clean test should pass, try changing your if statement on line 36 to
`if(i == j && transformationMatrix[i][j] != 1)` to test for the diagonal entries 
and then add an else if below it that says 
`else if( i != j && transformationMatrix[i][j] != 0 )` to test the non-diagonal entries